src/ditto/fx/utils.py

```diff
@@ -305,14 +306,15 @@ def find_nearest_node(
         follow_parents (bool, optional): Whether to follow parent nodes. Defaults to True
         follow_first_only (bool, optional): Whether to follow only the first node in the parent or child list.
             Defaults to False
+        num_limits (int, optional): The maximum number of nodes to search. Defaults to 10
```

Comment:
IMO asking whoever calling this function to provide an upper bound for the maximum number of nodes to search is counterintuitive.
How about limiting the loop iteration by maximum search depth? (**Note**: The default value must be amended as appropriate, for example, 3 or 5 looks reasonable for me.)
```suggestion
        max_depth (int, optional): The maximum depth to search. Defaults to 10
```

Reply:
I agree with that. I am fixing it by using maximum depth instead of a maximum number of nodes to search. It will be fixed in another PR.